### ISSUE TITLE
Auditory feedback

### ISSUE THREAD

**Issue by PeabodySam** [TheLegendOfMataNui/game-issues#169]
This is discussing the **beta** (though I'm sure someone with free time could apply this to the alpha).

Feedback is an important aspect of game design.  Without it, it's less clear to the player what's going on in the game.  This can come in visual and auditory flavors.

Visual feedback is simple enough: something happens, you see something.  For example, when a Toa hits a Rahi with an Energy Sphere, the Rahi flashes blue.  When a Rahi attacks a Toa and deals damage, the Toa flashes red.

I'm thinking this game could use more auditory feedback.  There is already some present in the game (Rahi nests play a specific sound when destroyed, Rahi make a certain sounded when defeated, etc.), but there are also a lot of silent actions that probably could use sound effects to provide auditory feedback.  If you closed your eyes, how could you tell a Rahi spawned from a nest and was attacking your Toa?  If an off-screen Rahi was shooting fireballs at you, how could you tell before you get hit?

For example, bats.  The only auditory feedback provided by bats is the sound effect they make when they are defeated; otherwise, they are completely silent.  However, the alpha (and presumably the beta, though I can't confirm) contains sound files for "batattack", "batattacknohit", and "bathit".  These could be useful as auditory feedback; "batattack" would alert players that a bat is attacking them, while "bathit" would let players know that their own attack successfully landed a hit.

Other similar sound files are "bugattack", "Bughit", "bugresurrect" (this could be used for Rahi spawning from nests), "crabattack", "scrppinch", "scrpsting", "spider hit", "spiderspitlong", "spiderspitshort", "ToaDamage", "Toadie", and "toaressurect".

This isn't just limited to fighting Rahi, though.  "maskbeginuse" was used in the alpha as auditory feedback letting players know that they just turned activated their Kanohi, but this auditory feedback was removed in the beta.  As obnoxious as it is to hear "MaskInUse" on loop (especially in Gali's stage), "maskbeginuse" could certainly be kept.  "nightvisionon" could be specific to the Ruru.

**Comment by ShadowOfNight** [comment]
@PeabodySam I recently began looking into this very issue.  Some of the sounds for bosses, for example, are hardcoded and called through the exe.  This makes adding sounds to the Vatuka-Nui extremely difficult.  Other sounds and events are called through the base.osi in conjunction with the soundtable.slb, which makes it easier to implement new sounds.  Others are called through the animevents.slb in each character's blk in tandem with the osi and sountable.  

In short, we know how the sounds are called, so there is a high probability we can get a lot of them functioning.  Case in point would be bat sounds.  

Hud sound effects like you mentioned in the final paragraph of your post would likely be the easiest to implement.
